src/actions/actions.ts

```diff
@@ -1900,94 +1908,168 @@ class ActionChangeChar extends BaseCommand {
 class MovementAWordTextObject extends BaseMovement {
   modes = [ModeName.Normal, ModeName.Visual];
   keys = ["a", "w"];
+  canBePrefixedWithCount = true;
 
   public async execAction(position: Position, vimState: VimState): Promise<IMovement> {
-    if (vimState.currentMode === ModeName.Visual && !vimState.cursorPosition.isEqual(vimState.cursorStartPosition)) {
-      // TODO: This is kind of a bad way to do this, but text objects only work in
-      // visual mode if you JUST entered visual mode
```

Comment:
I moved this piece of code to below https://github.com/VSCodeVim/Vim/pull/385/files#diff-7addcd1a8a241060a73b1fe4abb39960R1932 . If we entered Visual Mode and moved the cursor (selection length > 1), the final result selection depends on the cursor direction. So I checked `vimState.cursorPosition.isBefore(vimState.cursorStartPosition)`. So in my implementation, all cases are covered.